packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -45,6 +52,38 @@ interface EnqueueTaskParams {
   name: string;
 }
 
+/**
+ * A pino-based singleton logger for use within Google Cloud Functions.
+ * Provides a metric() method to log metrics-related data
+ */
+export class GCFLogger {
+  private static logger: pino.Logger;
+
+  /**
+   * Get the logger instance
+   * @returns a pino-based logger
+   */
+  public static get(): pino.Logger {
+    if (!this.logger) {
+      this.logger = this.initLogger();
+    }
+    return this.logger;
+  }
+
+  private static initLogger(dest?: pino.DestinationStream): pino.Logger {
+    const defaultOptions: pino.LoggerOptions = {
+      customLevels: {
+        metric: 30,
+      },
+      level: 'trace',
+    };
+    if (!dest) {
+      dest = pino.destination({sync: true});
+    }
+    return pino(defaultOptions, dest);
```

Comment:
I agree with @chingor13 on this topic, my advise would be to encapsulate `pino`, and expose an interface like:

```
logger.info()
logger.warn()
logger.error()
```

vs., returning an instance of pino.

Reply:
Yep, sounds good. I planned to do this initially but thought it might be too much wrapping, but since you and @chingor13 both seem to like this approach, I'll go ahead and do that. Makes for more robust code I think.